src/client/mapblock_mesh.cpp

```diff
@@ -49,7 +49,7 @@ void MeshMakeData::fillBlockDataBegin(const v3s16 &blockpos)
 
 	m_vmanip.clear();
 	VoxelArea voxel_area(blockpos_nodes - v3s16(1,1,1) * MAP_BLOCKSIZE,
```

Comment:
If side_length is anything but two this one also would have be adjusted.

Reply:
No, this is to build one-block layer of neighbours around the mesh. There is also +1 on all dimensions of the positive side.